v3/src/components/axis/hooks/use-sub-axis.ts

```diff
@@ -350,8 +351,8 @@ export const useSubAxis = ({
 
   // update d3 scale and axis when axis domain changes
   useEffect(function installDomainSync() {
-    const disposer = autorun(() => {
-      const _axisModel = axisProvider.getAxis?.(axisPlace)
+    const mobXAutorun = new MobXAutorun(() => {
+      const _axisModel = axisProvider?.getAxis?.(axisPlace)
       if (_axisModel && isAlive(_axisModel)) {
```

Comment:
Probably not important, but could you use `isAliveSafe` here? Change the line to `if (isAliveSafe(_axisModel)) {`?

Reply:
Good catch -- I wrote this part before I added `isAliveSafe()` and never revisited. I'll make the change.